packages/react-native-web/src/exports/Modal/ModalAnimation.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright (c) Nicolas Gallagher.
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import { useEffect, useState, useCallback, useMemo } from 'react';
+
+import StyleSheet from '../StyleSheet';
+import createElement from '../createElement';
+
+import type { ModalAnimationProps } from './types';
+
+
+const ANIMATION_DURATION = 300;
+
+function getAnimationStyle(animationType, visible) {
+  if (animationType === 'slide') {
+    return [
+      (visible ? styles.animatedIn : styles.animatedOut),
+      (visible ? styles.slideIn : styles.slideOut)
+    ]
```

Comment:
Not sure I understand - aren't they already?  They're created via `StyleSheet.create`  Do you have an example where this is done?

Or do you mean combining animatedIn & slideIn / animaedOut & slideOut, etc?

Reply:
I think they only work if they are applied dynamically 

> > Instead of trying to focus every element we could maybe use de data-focusable of react native web or is that too high level?
> 
> That's possible. Do you have a link to where that's documented? I'm having trouble finding it..
> 
> Would it work as expected to
> 
> * the W3C Accessibility Guidelines
> * aria browser hints
> * interactions with other focus management mechanisms (eg, TV UI)
> 
> If so, it might be worth exploring.
> 
> I'm always in the camp of "less is more". With this PR I was mostly trying to cut down on the amount of overridden UX interactions & let the browser handle as much as possible - so we aren't re-implementing what's already been solved by the browsers. :) Fewer cases for unexpected interactions.
> 
> For example, If someone _does_ want to make a TV UI with react native they'd still be able to - this feature does not override MUCH of the focus logic - only keeping the focus within the modal per W3CAG suggestion.


React Native Web adds a data-focusable=true on elements which are focusable. But I don't know if this will be forever.
https://github.com/necolas/react-native-web/pull/1698/files#diff-13924d58d5a7ddd791f0e9ae7b588c46R75

Reply:
We don't need to create new arrays each time, they can be defined outside the function:

```js
const animationSlideInStyles = [ styles.animatedIn, styles.slideIn ];
const animationSlideOutStyles = [ styles.animatedOut, styles.slideOut ];

function getAnimationStyle(animationType, visible) {
  if (animationType === 'slide') {
    return visible ? animationSlideInStyles : animationSlideOutStyles;
  }
  // ...
}
```

Reply:
Got it!  Makes much more sense that way. :) Thanks!